compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmInstructionGen.java

```diff
@@ -1530,6 +1534,56 @@ void generateTypeTestIns(BIRNonTerminator.TypeTest typeTestIns) {
         this.storeToVar(typeTestIns.lhsOp.variableDcl);
     }
 
+    private boolean canHandleErrorUnionType(BIRNonTerminator.TypeTest typeTestIns) {
+        var rhsVar = typeTestIns.rhsOp.variableDcl;
+        BType type = typeTestIns.type;
+        if (cannotOptimizeType(rhsVar, type)) {
```

Comment:
Shall we move this to generateTypeTestIns, so we can get rid of one if condition?

Reply:
We can make this function void as well.